backend/redis_client.py

```
@@ -14,21 +14,27 @@
+# Import timeout handling
+from core.timeout_handler import get_timeout_config
+import asyncio
+
 logger = logging.getLogger(__name__)
 
 
 class RedisClient:
     """Redis client for distributed coordination."""
 
-    def __init__(self, redis_url: str = "redis://localhost:6379"):
+    def __init__(self, redis_url: str = "redis://localhost:6379", timeout: float = None):
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)